building-and-deploying-a-fastapi-app.md

```diff
@@ -0,0 +1,541 @@
+                {"id":2,"name":"Donut","ingredients":["Flour","Milk","Butter"]},
+            ]
+}
+```
+
+You can also retrieve the recipe from its ID:
+
+```
+curl -X GET http://localhost:8080/recipe/2 -H 'Content-Type: application/json'
+```
+
+You will get the response:
+
+```json
+{
+    "data": [
+                {"id":2,"name":"Donut","ingredients":["Flour","Milk","Butter"]}
+            ]
+}
+```
+
+
+With the POST route in place, define the UPDATE route:
+
+
+
+```py
+@app.put("/recipe", tags=["Recipe"])
+def update_recipe(id: int, recipe_data: UpdateRecipeSchema) -> dict:
+    stored_recipe = {}
+    for recipe in recipes:
+        if recipe["id"] == id:
+            stored_recipe = recipe
+        else:
+            return {
+                "error": "No such recipe exist"
+            }
+
+    stored_recipe_model = RecipeSchema(**stored_recipe)
+    update_recipe = recipe_data.dict(exclude_unset=True)
+    updated_recipe = stored_recipe_model.copy(update=update_recipe)
+    recipes[recipes.index(stored_recipe_model)] = jsonable_encoder(updated_recipe)
+
+    return {
+        "message": "Recipe updated successfully."
+    }
+
+```
+
+In the code block above, you find a recipe by the ID supplied and perform a partial or full update depending on the content of the request body passed. If no such ID exists, it returns an error message. 
+
+Test the UPDATE route by changing thw title of the first recipe to __Buns__ using curl:
+
+```
+curl -X PUT "http://0.0.0.0:8080/recipe?id=1" -H  "accept: application/json" -H  "Content-Type: application/json" -d "{  \"name\": \"Buns\",}"
+```
+
+The UPDATE request returns this response:
+
+```json
+{
+  "message": "Recipe updated successfully."
+}
+```
+
+Let's define the final route, DELETE. The DELETE route is used to remove recipe in your application using their ID. Beneath the UPDATE route, add the following:
+
+```py
+@app.delete("/recipe/{id}", tags=["Recipe"])
+def delete_recipe(id: int) -> dict:
+    if id > len(recipes) or id < 1:
+        return {
+            "error": "Invalid ID passed"
+        }
+
+    for recipe in recipes:
+        if recipe['id'] == id:
+            recipes.remove(recipe)
+            return {
+                "message": "Recipe deleted successfully."
+            }
+
+    return {
+        "error": "No such recipe with ID {} exist".format(id)
+    }
+
+```
+
+In the code block above, you first check if the recipe whose ID is passed exists. If it does, the recipe is removed. Otherwise, an error message is returned. Test the DELETE route:
+
+```sh
+curl -X DELETE "http://0.0.0.0:8080/recipe/1" -H  "accept: application/json"
+```
+
+The request above return this response:
+
+```json
+{
+  "message": "Recipe deleted successfully."
+}
+```
+
+You have successfully built the CRUD application. 
+
+## Deploying to Okteto
+
+In this section, you will be deploying the application to Okteto. 
```

Comment:
Could we add a bit of why you'd want to do this? I can think of the following reasons:
1. To validate your application in a realistic environment
2. To share it with other people
3. To get an HTTPs endpoint

Reply:
Added a section.